### PR TITLE
Add edge attribute visualization

### LINKED ISSUE #182: Add edge attribute visualization
Similar to node attribute vis, the edge attribute vis will show a summary visualization for edge attributes

### PR DESCRIPTION
**Pull request by gotdairyya**
Enable edge visualizations
Closes #182 

Stacked bar chart for visualizing categorical attributes
Box and whisker plot for numerical attributes

### CODE REVIEW

**Comment by gotdairyya** [comment]
@JackWilb I need your feedback on architecture.
Right now the edge attributes only visualize if the node attribute has been toggled first.

**Review by JackWilb — COMMENTED** [review]
On the whole, I'm happy with the changes. I noticed a bug, though, that means you can't have both link and node attributes displayed at the same time. It could be problems with positioning, or the rendering in general; it will take some exploration to see what's causing it.

**Comment by gotdairyya** [comment]
@JackWilb can you take a look at the recent commit? There are two bugs that I need some help on

- [ ] Fixing categorical edge vis (it draws everything black, data is right though)
- [ ] Currently only works if node attribute is selected first

**Comment by gotdairyya** [comment]
@JackWilb I fixed most of the type errors but there are a couple I couldn't figure out. Could you take a look?

File: src/components/MultiMatrix/MultiMatrix.vue [hunk 1]
```diff
@@ -445,16 +530,61 @@ export default Vue.extend({
...
+        .concat(
+          this.visualizedAttributes.filter((x) => !linkDifference.includes(x)),
+        );
+      if (this.combinedAttributes.includes(linkDifference[0])) {
+        this.combinedAttributes = this.combinedAttributes.filter(
+          (el: string) => el != linkDifference[0],
+        );
+      } else {
+        this.combinedAttributes.push(linkDifference[0]);
+      }
```
Comment (JackWilb) [inline src/components/MultiMatrix/MultiMatrix.vue:585]: Same as 2 above. Can we expand the array here to get them all?

**Review by JackWilb — COMMENTED** [review]
Seems to work well. I'm less familiar with the nesting and the data processing, but everything else seems good. I had a couple suggestions for how we might make the code a little cleaner, let me know if they seem reasonable or if you think we should defer them to the future.

File: src/components/MultiMatrix/MultiMatrix.vue [hunk 2]
```diff
@@ -234,6 +252,37 @@ export default Vue.extend({
...
     },
 
+    rowData(): any {
+      const rowData = nest()
+        .key((d: any) => d._from)
+        .entries(this.network.edges);
+
+      console.log(rowData);
+
+      const edgeAttributes = Object.keys(rowData[0].values[0]);
```
[the remark this answers is not in the document]
Comment (gotdairyya) [reply src/components/MultiMatrix/MultiMatrix.vue:262]: should I resolve this?

File: src/components/MultiMatrix/MultiMatrix.vue [hunk 3]
```diff
@@ -234,6 +252,37 @@ export default Vue.extend({
       return computedIdMap;
     },
 
+    rowData(): any {
+      const rowData = nest()
```
[the remark this answers is not in the document]
Comment (gotdairyya) [reply src/components/MultiMatrix/MultiMatrix.vue]: Resolved with commit 1d7489994e52a06798d91d46e857af42bef81083

**Comment by JackWilb** [comment]
I just want to leave a note for myself that I need take a quick look at why the column widths aren't working as we intend

File: src/components/MultiMatrix/MultiMatrix.vue [hunk 4]
```diff
@@ -445,16 +530,61 @@ export default Vue.extend({
...
+        .concat(
+          this.visualizedLinkAttributes.filter(
+            (x) => !this.combinedAttributes.includes(x),
+          ),
+        );
+      linkDifference = linkDifference
+        .filter((x) => !this.visualizedAttributes.includes(x))
+        .concat(
+          this.visualizedAttributes.filter((x) => !linkDifference.includes(x)),
+        );
```
[the remark this answers is not in the document]
Comment (gotdairyya) [reply src/components/MultiMatrix/MultiMatrix.vue:578]: Should be resolved with 3bd37863326ed5c14ba0631be2cdaf07d6d43f96

File: src/components/MultiMatrix/MultiMatrix.vue [hunk 5]
```diff
@@ -445,16 +530,61 @@ export default Vue.extend({
...
+            (x) => !this.combinedAttributes.includes(x),
+          ),
+        );
+      nodeDifference = nodeDifference
+        .filter((x) => !this.visualizedLinkAttributes.includes(x))
+        .concat(
+          this.visualizedLinkAttributes.filter(
+            (x) => !nodeDifference.includes(x),
+          ),
+        );
```
[the remark this answers is not in the document]
Comment (gotdairyya) [reply src/components/MultiMatrix/MultiMatrix.vue:555]: Should be resolved with 3bd37863326ed5c14ba0631be2cdaf07d6d43f96

File: src/components/MultiMatrix/MultiMatrix.vue [hunk 6]
```diff
@@ -445,16 +530,61 @@ export default Vue.extend({
...
+      if (this.combinedAttributes.includes(nodeDifference[0])) {
+        this.combinedAttributes = this.combinedAttributes.filter(
+          (el: string) => el != nodeDifference[0],
+        );
+      } else {
+        this.combinedAttributes.push(nodeDifference[0]);
+      }
```
[the remark this answers is not in the document]
Comment (gotdairyya) [reply src/components/MultiMatrix/MultiMatrix.vue:562]: nodeDifference is a list that takes the difference in the nodes selected and then the resulting list will always only have 1 element in it, which is why there is the [0]. I did update line 562 with your suggestion

File: src/components/MultiMatrix/MultiMatrix.vue (hunk 1 above)
Comment (gotdairyya) [reply src/components/MultiMatrix/MultiMatrix.vue]: Should be addressed with commit 4f7a7eb38ea8c9c65854b020764007d825232308

**Comment by gotdairyya** [comment]
> I just want to leave a note for myself that I need take a quick look at why the column widths aren't working as we intend

I think that a new column for the # of children is being added every time a new attribute is added (taking up more space each time)

File: src/components/MultiMatrix/MultiMatrix.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (JackWilb) [reply src/components/MultiMatrix/MultiMatrix.vue]: Yep, looks like you did :)

**Review by JackWilb — APPROVED** [review]

**Pull request merged**